src/main/java/io/jenkins/plugins/artifact_manager_s3/JCloudsArtifactManager.java

```diff
@@ -349,6 +349,7 @@ private static void uploadFile(Path f, URL url) throws IOException {
         HttpURLConnection connection = (HttpURLConnection) url.openConnection();
         connection.setDoOutput(true);
         connection.setRequestMethod("PUT");
+        connection.setFixedLengthStreamingMode(Files.size(f)); // prevent loading file in memory
         try (OutputStream out = connection.getOutputStream()) {
```

Comment:
Note that this could now produce a `HttpRetryException`. Not an issue for S3 I suppose.